src/lighteval/tasks/lighteval_task.py

```diff
@@ -504,7 +507,19 @@ def get_metric_method_from_category(self, metric_category):
         if not self.has_metric_category[metric_category]:
             raise ValueError(f"Requested a metric category {metric_category} absent from the task list.")
 
-        return LightevalTask._get_metric_method_from_category(metric_category)
+        metric_method = LightevalTask._get_metric_method_from_category(metric_category)
+        # Bad hack. I had no other way.
```

Comment:
I suggest considering task as the argument of `apply_*_metrics` functions as a workaround.